spec/models/tag_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+describe Tag do
+
+  it "decreases tag_count when a debate is hidden" do
+    debate = create(:debate)
+    tag = create(:tag)
+    tagging = create(:tagging, tag: tag, taggable: debate)
+
+    expect(tag.taggings_count).to eq(1)
+
+    debate.update(hidden_at: Time.now)
+
+    tag.reload
+    expect(tag.taggings_count).to eq(0)
+  end
+
+  it "decreases tag_count when a proposal is hidden" do
```

Comment:
We don't use shared examples in the whole project. Right now we just leave with repetition in specs (It's surprisingly easy to leave with it).

We're not however closed to adding them. We just never considered doing so.

I would however not include them just to improve these two specs alone - I'm not sure the code repetition it would save would be worth it, compared to the extra complexity of having shared examples here .

That said, if shared examples could be used in to clean the worst offenders, then that would change my mind. The specs where proposals, debates and/or comments are together are usually the worst offenders, repetition-wise. For example:

https://github.com/consul/consul/blob/master/spec/features/comments/debates_spec.rb
https://github.com/consul/consul/blob/master/spec/features/comments/proposals_spec.rb

If you find a way to share code between those two and halve that code, I would gladly accept a pr for that.